src/app/sidenav/sidenav.component.html

```diff
@@ -1,22 +1,42 @@
-<mat-sidenav-container fullscreen [style.top.px]="utils.screenWidth <= 599 ? 56 : 64">
-
-  <mat-sidenav #sidenav [mode]="utils.isMobile ? 'over': 'side'" [opened]="!utils.isMobile" [style.width.px]="sidebarWidth">
+<mat-sidenav-container fullscreen [style.top.px]="utils.screenWidth <= 599 ? 56 : 64">     
+  <mat-sidenav #sidenav [mode]="utils.isMobile ? 'over': 'side'" [opened]="!utils.isMobile && sidebarState" [style.width.px]="sidebarWidth">
     <mat-nav-list>
       <a mat-list-item [routerLink]="['/create-community']">Create Community</a>
       <mat-divider></mat-divider>
-      <mat-list-item *ngFor="let communityFav of communitiesFavourited; let j = index">
-        <a mat-line *ngIf="communityFav.isFavourited" [routerLink]="['/']">
-           {{ communityFav.communityname }}
-        </a>
-        <button mat-icon-button (click)="toggleFavourite(communityFav)" ><mat-icon class="Favourited">star</mat-icon></button>
-      </mat-list-item>
+      <mat-toolbar *ngIf="userLoggedIn || this.userService.isLoggedIn()">
+        Favourited Communities
+      </mat-toolbar>
```

Comment:
The styling for this is insanely huge

Reply:
True; I can reduce it. I think a MatExpansion could be useful for grouping the favourite and top communities together.

Reply:
Yes, it would be an automatically expanded mat expansion with top communities under it. Users can chose to hide it if they just want to focus on their favourited communities.

Reply:
If you've got the time for it, sure